maven-extension/src/main/java/io/opentelemetry/maven/SpanRegistry.java

```diff
@@ -114,9 +126,11 @@ public Span removeSpan(MojoExecution mojoExecution) {
 
     abstract String artifactId();
 
-    public static MavenProjectKey fromMavenProject(@Nonnull MavenProject mavenProject) {
+    abstract String version();
```

Comment:
Where is `version()` used?

Reply:
It's a protection for the map of spans by MojoExecutionKeys. I imagine 2 mojo executions can't execute simultaneously  on the same project with the different version of the Maven plugin but I felt it's a safe guardrail.